src/Symfony/Component/Translation/Bridge/PoEditor/Tests/PoEditorProviderTest.php

```diff
@@ -0,0 +1,514 @@
+                            'term' => 'a',
+                            'context' => 'messages',
+                            'translation' => ['content' => 'trans_en_a'],
+                        ],
+                        [
+                            'term' => 'post.num_comments',
+                            'context' => 'validators',
+                            'translation' => ['content' => '{count, plural, one {# comment} other {# comments}}'],
+                        ],
+                    ]),
+                ]), $options['body']);
+
+                return $successResponse;
+            },
+            'addTranslationsFr' => function (string $method, string $url, array $options = []) use ($successResponse): MockResponse {
+                $this->assertSame('POST', $method);
+                $this->assertSame(http_build_query([
+                    'api_token' => 'API_KEY',
+                    'id' => 'PROJECT_ID',
+                    'language' => 'fr',
+                    'data' => json_encode([
+                        [
+                            'term' => 'a',
+                            'context' => 'messages',
+                            'translation' => ['content' => 'trans_fr_a'],
+                        ],
+                        [
+                            'term' => 'post.num_comments',
+                            'context' => 'validators',
+                            'translation' => ['content' => '{count, plural, one {# commentaire} other {# commentaires}}'],
+                        ],
+                    ]),
+                ]), $options['body']);
+
+                return $successResponse;
+            },
+        ];
+
+        $translatorBag = new TranslatorBag();
+        $translatorBag->addCatalogue(new MessageCatalogue('en', [
+            'messages' => ['a' => 'trans_en_a'],
+            'validators' => ['post.num_comments' => '{count, plural, one {# comment} other {# comments}}'],
+        ]));
+        $translatorBag->addCatalogue(new MessageCatalogue('fr', [
+            'messages' => ['a' => 'trans_fr_a'],
+            'validators' => ['post.num_comments' => '{count, plural, one {# commentaire} other {# commentaires}}'],
+        ]));
+
+        $provider = $this->createProvider(
+            new MockHttpClient($responses, 'https://api.poeditor.com/v2/'),
+            $this->getLoader(),
+            $this->getLogger(),
+            $this->getDefaultLocale(),
+            'api.poeditor.com/v2/'
+        );
+
+        $provider->write($translatorBag);
+    }
+
+    /**
+     * @dataProvider getResponsesForOneLocaleAndOneDomain
+     */
+    public function testReadForOneLocaleAndOneDomain(string $locale, string $domain, string $responseContent, TranslatorBag $expectedTranslatorBag)
+    {
+        $loader = $this->getLoader();
+        $loader->expects($this->once())
+            ->method('load')
+            ->willReturn((new XliffFileLoader())->load($responseContent, $locale, $domain));
+
```

Comment:
I'm not happy with this, but I don't find the correct way to either test it, or mock it. cc @nicolas-grekas  (same in `testReadForManyLocalesAndManyDomains`)

Reply:
see deps=low failures